posts/2023/installing-farmos-on-ddev-with-composer/index.md

```diff
@@ -0,0 +1,105 @@
+
+DDEV's many features are what make it a great way to build Drupal and farmOS
+projects in a local environment:
+
+* Cross-platform, open source software.
+* Supports many versions of Docker or can be used with Colima.
+* When combined with Colima and Mutagen, performance is significantly improved.
+* Automatic HTTPS.
+* Run multiple projects concurrently.
+* Can be used to develop many other CMSs.
+* Support multiple databases (including MySQL, Postgre, SQLite).
+* Lots of developer tools included, and is easy to extend and integrate.  
+
+# Installing farmOS with DDEV and Composer
+
+- First, follow the instructions to install and setup DDEV: https://ddev.readthedocs.io/en/latest/users/install/
+- Once installed, the following commands will create a DDEV project and install farmOS:
+
+```
+ddev config --project-type=drupal10 --docroot=web --create-docroot --database=postgres:15 --php-version=8.1
+ddev get wotnak/ddev-php-geos
+ddev start
+ddev composer create "farmos/project:3.x-dev"
+ddev drush site:install farm --account-name=admin --account-pass=admin -y
+ddev drush uli
+ddev launch
+```
+
+## Usage
+
+### Useful DDEV commands:
+- Start/stop a project: `ddev start`/ `ddev stop`
+- Open project: `ddev launch`
+- Stop ddev: `ddev poweroff`
+- PHP CLI: `ddev php`
+- Additional DDEV commands: https://ddev.readthedocs.io/en/latest/users/usage/cli/
+
+### Drush
+- Drush commands: `ddev drush {drush command}`
+- Login link: `ddev drush uli`
+- CRON: `ddev drush cron`
+- Additional Drush commands: https://www.drush.org/latest/commands/all/
+
+### Composer:
+- Add dependency (exmple `farm_organic`): `ddev composer require drupal/farm_organic`
+- Updating dependencies: `ddev composer update`
+- See farmOS documentation on using Composer: https://farmos.org/hosting/composer/
+
+# Going to production
+
+It should be noted that DDEV _is not_ designed to be used as a production
+hosting environment. DDEV is designed to make managing a local development
+environment easier. Taking a local environment to production depends entirely
+on how that production environment is built. Most composer-based workflows
+will expect the project's `composer.json`, `composer.lock` and any application
+code be checked in to source control, from which the production environment
+can be built.
+
+DDEV provides integrations with hosting providers to simplify the process of
+syncing production data into a local environment. For more information see
+[Hosting Provider Integration](https://ddev.readthedocs.io/en/stable/users/providers/)
+
+# Docker image limitations
+
+farmOS has been using Docker and Docker Compose to create and distribute
+reproducible development environments for the past many years. Each commit to
```

Comment:
"past many years" is a bit awkward. How about;

```suggestion
farmOS has historically used Docker and Docker Compose to create and distribute
reproducible development environments. Each commit to
```

Reply:
Hmm yes - but does "historically used" make it sound like this is no longer the case? Maybe simply:

```suggestion
farmOS uses Docker and Docker Compose to create and distribute reproducible
development environments. Each commit to
```